### PR TITLE
Copy lrcTokenAddress to a local variable.

### PR DESCRIPTION
**Pull request by BenjaminPrice**
By copying this to a local variable we can avoid reading from storage multiple times and instead read from the cheaper memory.

### CODE REVIEW

**Comment by dong77** [comment]
I haven't test it ,but doesn't this change introduce additional gas saving? I thought compiler optimization would do the job.

**Comment by BenjaminPrice** [comment]
Surprisingly, in my local tests, it does make a saving. If you convert the compiled contract to OPCODE you can see the SLOAD commands multiple times. The compiler isn't being smart enough and storing it in memory after the first read. Maybe things changed in an updated patch version of Solidity but it should be worth testing to confirm.

File: contracts/LoopringProtocolImpl.sol [hunk 1]
```diff
@@ -477,11 +477,12 @@ contract LoopringProtocolImpl is LoopringProtocol {
         // Calculate each order's `lrcFee` and `lrcRewrard` and splict how much
         // of `fillAmountS` shall be paid to matching order or miner as margin
         // split.
-
-        calculateRingFees(delegate, ring);
+        address lrcToken = lrcTokenAddress;
```
Comment (dong77) [inline contracts/LoopringProtocolImpl.sol]: how about renaming to "_lrcTokenAddress"?

**Comment by kongliangzhong** [comment]
I have tested this PR, it saved about 450 gas.

File: contracts/LoopringProtocolImpl.sol (hunk 1 above)
Comment (BenjaminPrice) [reply contracts/LoopringProtocolImpl.sol]: Renamed

**Comment by BenjaminPrice** [comment]
When resolving the merge conflicts, I made the variable rename request to _lrcTokenAddress

**Comment by dong77** [comment]
reduced gas from 400427 to 399956.

**Comment by BenjaminPrice** [comment]
Please add to `optimization_submissions.md`

**Pull request merged**